### PR TITLE
fix(docs-infra): improve the layout/appearance of the "Press kit" page

### PR DESCRIPTION
**Pull request by gkalpak**
_**NOTE:**_
_Along with #40427, #40704, #40802, #40881 and #40944, this PR concludes the effort to clean up and bring in the changes from #36045. Huge shout-out to @sjtrimble for the original work ❤️_

##
Some layout/appearance improvements for the "Press kit" page:

<details>
  <summary><b>fix(docs-infra): improve the layout and appearance of the "Press kit" page</b> <i>(click to expand)</i></summary><br />

  This commit improves the layout and appearance of the "Press kit" page. Most notable changes include:
  - Center content and limit max width to 50em (which is 800px by default).
  - Remove uppercase transform for headings.
  - Use a two-column layout for icons (instead of one icon per row).
  - Show "download" icons next to download links.

  _**Before (wide):**_
  <p align="center">
    presskit page (wide) before
  </p>

  _**Before (narrow 1/2):**_
  <p align="center">
    presskit page (narrow 1/2) before
  </p>

  _**Before (narrow 2/2):**_
  <p align="center">
    presskit page (narrow 2/2) before
  </p>

  _**After (wide):**_
  <p align="center">
    presskit page (wide) after
  </p>

  _**After (narrow 1/2):**_
  <p align="center">
    presskit page (narrow 1/2) after
  </p>

  _**After (narrow 2/2):**_
  <p align="center">
    presskit page (narrow 2/2) after
  </p>

  This change is based on code from #36045.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

---
<details>
  <summary><b>fix(docs-infra): more clearly separate different sections on the "Press kit" page</b> <i>(click to expand)</i></summary><br />

  This commit adds a horizontal line and some extra spacing between different sections of the "Press kit" page to make it easier to identify where one section ends and the next begins.

  _**Before:**_
  <p align="center">
    presskit sections before
  </p>

  _**After:**_
  <p align="center">
    presskit sections after
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): add links to glossary for terms on the "Press kit" page</b> <i>(click to expand)</i></summary><br />

  The "Press kit" page has a "Terminology" section that lists a couple of terms (currently AOT and JIT compilation).

  This commit adds links to the corresponding entries in "Glossary" to make it easier for people to find out what the terms mean.

</details>

### CODE REVIEW

**Comment by sjtrimble** [comment]
@googlebot I consent

File: aio/src/styles/2-modules/_presskit.scss [hunk 1]
```diff
@@ -7,6 +7,12 @@
   }
 
   .presskit-section {
+    &:not(:first-child) {
+      border-top: 1px solid $lightgray;
+      margin-top: 4rem;
+      padding-top: 2rem;
+    }
+
```
Comment (petebacondarwin) [inline aio/src/styles/2-modules/_presskit.scss]: WOT NO `<hr><br>` rather than adding styles 😛 ?

**Review by petebacondarwin — APPROVED** [review]
LGTM!

File: aio/src/styles/2-modules/_presskit.scss (hunk 1 above)
Comment (gkalpak) [reply aio/src/styles/2-modules/_presskit.scss]: Keeping it DRY :wink:
(The one in "Events" was a single occurrence 😛)

**Review by aikithoughts — APPROVED** [review]
Reviewed-for: global-docs-approvers

**Comment by Splaktar** [comment]
This looks so much better! Thanks!

**Pull request closed without merging**